### ISSUE TITLE
The outcome of the detection depicts the entire image as human, but it does not align with reality.

### ISSUE THREAD

**Issue by FearlessArmy** [Deci-AI/super-gradients#1477]
### 💡 Your Question

**I utilized yolonas to train a single-class detection model on COCO and LVIS datasets, and employed this model for prediction. Surprisingly, within a subset of the MOT dataset, there was an amusing occurrence where an entire frame was mistakenly identified as containing humans. the confidence level even reached as high as 0.65. However, through some preliminary attempts on other datasets, such comical results were not observed.** 
![image](https://github.com/Deci-AI/super-gradients/assets/93211004/80105d44-e620-44d8-8157-1f4dd99ab1ec)
**The training script I executed is as follows:**
```
import os

import requests
import torch
from PIL import Image

from super_gradients.common.environment.device_utils import DeviceConfig
from super_gradients.training import Trainer, dataloaders, models
from super_gradients.training.dataloaders.dataloaders import (
    coco_detection_yolo_format_train, coco_detection_yolo_format_val
)
from super_gradients.training.losses import PPYoloELoss
from super_gradients.training.metrics import DetectionMetrics_050
from super_gradients.training.models.detection_models.pp_yolo_e import (
    PPYoloEPostPredictionCallback
)
from super_gradients.training.utils.distributed_training_utils import setup_device
setup_device(num_gpus=1)
class config:
    # trainer params
    CHECKPOINT_DIR = 'checkpoints'  # specify the path you want to save checkpoints to
    EXPERIMENT_NAME = 'cars-from-above'  # specify the experiment name

    # dataset params
    DATA_DIR = 'D:/data'  # parent directory to where data lives

    TRAIN_IMAGES_DIR = 'images/train'  # child dir of DATA_DIR where train images are
    TRAIN_LABELS_DIR = 'labels/train'  # child dir of DATA_DIR where train labels are

    VAL_IMAGES_DIR = 'images/test'  # child dir of DATA_DIR where validation images are
    VAL_LABELS_DIR = 'labels/test'  # child dir of DATA_DIR where validation labels are

    # TEST_IMAGES_DIR = 'images/test'  # child dir of DATA_DIR where validation images are
    # TEST_LABELS_DIR = 'labels/test'  # child dir of DATA_DIR where validation labels are
    CLASSES = ['pedestrian']  # 指定类名

    NUM_CLASSES = len(CLASSES) # 获取类个数

    # dataloader params - you can add whatever PyTorch dataloader params you have
    # could be different across train, val, and test
    DATALOADER_PARAMS = {
        'batch_size': 64,
        'num_workers': 18
    }

    # model params
    MODEL_NAME = 'yolo_nas_s'  # 可以选择 yolo_nas_s, yolo_nas_m, yolo_nas_l。分别是 小型，中型，大型
    PRETRAINED_WEIGHTS = 'coco'  # only one option here: coco


trainer = Trainer(experiment_name=config.EXPERIMENT_NAME, ckpt_root_dir=config.CHECKPOINT_DIR)

# 指定训练数据
train_data = coco_detection_yolo_format_train(
    dataset_params={
        'data_dir': config.DATA_DIR,
        'images_dir': config.TRAIN_IMAGES_DIR,
        'labels_dir': config.TRAIN_LABELS_DIR,
        'classes': config.CLASSES
    },
    dataloader_params=config.DATALOADER_PARAMS
)

# 指定评估数据
val_data = coco_detection_yolo_format_val(
    dataset_params={
        'data_dir': config.DATA_DIR,
        'images_dir': config.VAL_IMAGES_DIR,
        'labels_dir': config.VAL_LABELS_DIR,
        'classes': config.CLASSES
    },
    dataloader_params=config.DATALOADER_PARAMS
)

# test_data = coco_detection_yolo_format_val(
#     dataset_params={
#         'data_dir': config.DATA_DIR,
#         'images_dir': config.TEST_IMAGES_DIR,
#         'labels_dir': config.TEST_LABELS_DIR,
#         'classes': config.CLASSES
#     },
#
dataloader_params=config.DATALOADER_PARAMS
# )
# train_data.dataset.plot()

model = models.get(config.MODEL_NAME,
                   num_classes=config.NUM_CLASSES,
                   pretrained_weights=config.PRETRAINED_WEIGHTS
                   )
train_params = {
    # ENABLING SILENT MODE
    "average_best_models":True,
    "warmup_mode": "linear_epoch_step",
    "warmup_initial_lr": 1e-6,
    "lr_warmup_epochs": 3,
    "initial_lr": 2e-4,
    "lr_mode": "cosine",
    "cosine_final_lr_ratio": 0.1,
    "optimizer": "Adam",
    "optimizer_params": {"weight_decay": 0.0001},
    "zero_weight_decay_on_bias_and_bn": True,
    "ema": True,
    "ema_params": {"decay": 0.9, "decay_type": "threshold"},
    # ONLY TRAINING FOR 10 EPOCHS FOR THIS EXAMPLE NOTEBOOK
    "max_epochs": 400,
    "save_ckpt_epoch_list":[50,100,150,200, 250,300,350],
    "mixed_precision": True,
    "loss": PPYoloELoss(
        use_static_assigner=False,
        # NOTE: num_classes needs to be defined here
        num_classes=config.NUM_CLASSES,
        reg_max=16
    ),
    "valid_metrics_list": [
        DetectionMetrics_050(
            score_thres=0.1,
            top_k_predictions=300,
            # NOTE: num_classes needs to be defined here
            num_cls=config.NUM_CLASSES,
            normalize_targets=True,
            post_prediction_callback=PPYoloEPostPredictionCallback(
                score_threshold=0.01,
                nms_top_k=1000,
                max_predictions=300,
                nms_threshold=0.7
            )
        )
    ],
    "metric_to_watch": 'mAP@0.50',
    "greater_metric_to_watch_is_better":True
}

trainer.train(model=model,
              training_params=train_params,
              train_loader=train_data,
              valid_loader=val_data)

best_model = models.get(config.MODEL_NAME,
                        num_classes=config.NUM_CLASSES,
                        checkpoint_path=os.path.join(config.CHECKPOINT_DIR, config.EXPERIMENT_NAME, 'average_model.pth'))
```
**The testing methodology employed was:**
```
best_pth = "D:/yolonas/merdata_coco/ckpt_best.pth"
best_model = models.get("yolo_nas_s", num_classes=num_classes, checkpoint_path=best_pth).to(device)
best_model.predict(imgdir,iou=0.6, conf=0.3,fuse_model=False).show()
```
Here are the testing results on other MOT datasets:
![image](https://github.com/Deci-AI/super-gradients/assets/93211004/cdfe3168-da02-486d-8269-5c98842c7a54)

I am unable to comprehend the occurrence of this phenomenon and seek your assistance. I extend my gratitude in advance for your help.

### Versions

I conducted the training on a Linux system and performed the testing on both Windows and Linux systems. However, I found that the issue is not related to the system versions.

**Comment by BloodAxe** [comment]
Hi. Thanks for raising this up. I have a few questions regarding your training results.
1) What was the best mAP50 score did you get when training and how did it compare with other models you've tried?

2) mAP usually is not great metric at judging how good a model at FP. 
Instead, you can observe F1 score as follows:
```
    "valid_metrics_list": [
        DetectionMetrics_050(
            score_thres=0.1,
            top_k_predictions=300,
            # NOTE: num_classes needs to be defined here
            num_cls=config.NUM_CLASSES,
            normalize_targets=True,
            calc_best_score_thresholds=True, # !!! <- Important to enable to find the optimal confidence threshold that maximize F1
            post_prediction_callback=PPYoloEPostPredictionCallback(
                score_threshold=0.01,
                nms_top_k=1000,
                max_predictions=300,
                nms_threshold=0.7
            )
        )
    ],
```

And in `metric_to_watch` put `F10.5` instead of `mAP`.

3) If your training images are rectangular that are far from square, default dataset transformations may be not the optimal choice for your. You may want to change the `train_dataset_params.input_dim` (rows, cols) to size that is closer to your dataset.

`dataset_params={ ..., input_dim=(720, 1280) }` # Just an example

4) You can also employ [ExtremeBatchDetectionVisualizationCallback](https://github.com/Deci-AI/super-gradients/blob/master/src/super_gradients/training/utils/callbacks/callbacks.py#L1198) to see what is happening with predictions during training.

**Comment by Louis-Dupont** [comment]
@FearlessArmy I'm closing this issue due to inactivity. If you are still facing the issue feel free to re-open it with extra information.
